packages/edit-widgets/src/store/reducer.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { keyBy, mapValues, pick } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { combineReducers } from '@wordpress/data';
+
+export function widgetAreas( state = {}, action ) {
```

Comment:
Do we really need to keep the widgets areas in a store like this one?

I always wanted to move the `currentPost` object out of the `editor` store and just use `core-data` instead. 
For example, to update the widgets area, we could just call `saveEntityRecord` directly.

Anything I'm missing? It seems like it could be straightforward and simplifies this PR a lot.

Reply:
Hi @youknowriad, I followed your suggestion 👍.  I updated the code to use wp.data.dispatch('core').saveWidgetArea (saveEntityRecord ) instead of reimplementing parts of this logic. The code was simplified in this process.